### ISSUE TITLE
Extract.R line 22 error

### ISSUE THREAD

**Issue by gdbassett** [jimhester/lintr#272]
![error](https://user-images.githubusercontent.com/1392552/31082946-dbdf2b56-a755-11e7-801a-4a80ea7c05ae.png)

The error isn't verbose enough for me to know which of my files is causing it unfortunately.  Let me know how I can help figure out the cause.

Gabe

**Comment by jimhester** [comment]
1. Provide a link to the package that causes the error.
2. Provide the full results of running `traceback()` after the error occurs.
3. Run 
```r
for (file in list.files("R", full.names = T)) {
  cat(file)
  lintr::lint(file)
}
```

**Comment by gdbassett** [comment]
how the hell did I close this?  That's what I get for browsing on my phone.

1. The package is private unfortunately

2. 
```
> lintr::lint_package()
...Error in rep.int(NA_character_, max(ends - 1)) : invalid 'times' value
In addition: Warning message:
In max(ends - 1) : no non-missing arguments to max; returning -Inf
> traceback()
11: extract_r_source(source_file$filename, source_file$lines)
10: get_source_expressions(filename)
9: lint(file, ..., parse_settings = FALSE)
8: FUN(X[[i]], ...)
7: lapply(files, function(file) {
       if (interactive()) {
           message(".", appendLF = FALSE)
       }
       lint(file, ..., parse_settings = FALSE)
   })
6: inherits(x, class)
5: assign_item(x)
4: flatten_list(x, class = "lint")
3: structure(flatten_list(x, class = "lint"), class = "lints")
2: flatten_lints(lapply(files, function(file) {
       if (interactive()) {
           message(".", appendLF = FALSE)
       }
       lint(file, ..., parse_settings = FALSE)
   }))
1: lintr::lint_package()
```
3.
```
> for (file in list.files("R", full.names = T)) {
+     cat(paste0(file, "\n"))
+     lintr::lint(file)
+ }
<file names - gabe>
R/sysdata.rda
Error in substring(string, starts, ends) : 
  invalid multibyte string at '<binary gibberish - gabe>'
In addition: There were 24 warnings (use warnings() to see them)
```
Looks like I have a data file in my code directory that I wasn't aware of.

**Comment by russHyde** [comment]
This error used to be caused by this line in `lintr:::extract_r_source`:
```
output <- rep.int(NA_character_, max(ends - 1))
```
In the CRAN `lintr` 1.0.2, there was no check that `ends` might be empty, so if you passed an R/Rmd file that had no code-blocks in it, you effectively ended up running `max(c() - 1)`; which fails as above.

The number of code-block starts and ends are checked as of https://github.com/jimhester/lintr/commit/7fe7e9093c7e9f41eebe600d89a98375d68f8f40

Since no reprex was found, and I'm fairly certain this has been fixed I'm closing this issue
